### ISSUE TITLE
"No source available" in Text Explorer

### ISSUE THREAD

**Issue by kervinpierre** [JohnnyHendriks/TestAdapter_Catch2#33]
After quite a fight I can see tests listed in Test Explorer.

The problem know is that I can't click on a test in test explorer to navigate to the source code.  A very useful feature.

![image](https://user-images.githubusercontent.com/350213/67604796-94f16780-f74a-11e9-8bc7-1c16a427a895.png)

How can I get the source code to the tests listed in test explorer?  Is this a bug or missing feature?

**Comment by JohnnyHendriks** [comment]
This should work most of the time. There are however a few unhappy cases where it doesn't work. Looks like you hit upon some of those. I assume you are using the latest version of the test adapter and that you use the default test discovery feature.

The problem with default discovery is that the information retrieved from the Catch2 test executable may be split over several lines. Here some information may get lost (typically spaces). So if you have long path names to the cpp-files they may not get reconstructed correctly. If you add the  following setting to you runsettings-file (or if you already have it change the value of the setting to `debug`).

```
<Catch2Adapter>
  <!-- You should keep your other settings -->
  <Logging>debug</Logging>
</Catch2Adapter>
```

If you trigger a new discovery of the test, you will see additional information about the discovery process in the Visual Studio output window (you may have to manually select the output for `Tests`, typically it is set to `Build`). Note, in VS2019 you may have to switch to a different Configuration first before building one of your tests.

I'm already working on making the reconstruction of information for default discovery more robust. This should arrive in the next release of the **Test Adapter for Catch2**.

**Comment by kervinpierre** [comment]
Here's the output.  I can't find anything wrong there, and the output seems to have the file and line number.

`[2019-11-12 5:27:27.484 PM Informational] ========== Discovery skipped: All test containers are up to date ==========
[2019-11-12 5:27:27.512 PM Informational] ---------- Run started ----------
[2019-11-12 5:27:28.102 PM Informational] Started Catch2Adapter test execution...
[2019-11-12 5:27:28.104 PM Informational] Run test: CmdLine.V8. Simple Display help
[2019-11-12 5:27:28.104 PM Informational] Source: H:\test\myxxxxxxxxx-service\build\msvc\Debug\EP_CPP_TEST_main.exe
[2019-11-12 5:27:28.105 PM Informational] SolutionDirectory: H:\test\myxxxxxxxxx-service\
[2019-11-12 5:27:28.105 PM Informational] TestRunDirectory: H:\test\myxxxxxxxxx-service\TestResults
[2019-11-12 5:27:28.158 PM Informational] Executor log:
Source for test case: H:\test\myxxxxxxxxx-service\build\msvc\Debug\EP_CPP_TEST_main.exe
Commandline arguments used to run tests case: "CmdLine.V8. Simple Display help" --reporter xml --durations yes
Run test case: CmdLine.V8. Simple Display help
<?xml version="1.0" encoding="UTF-8"?>
<Catch name="EP_CPP_TEST_main.exe" filters="CmdLine.V8. Simple Display help">
  <Group name="EP_CPP_TEST_main.exe">
    <TestCase name="CmdLine.V8. Simple Display help" filename="H:\test\myxxxxxxxxx-service\tests\EP_CPP_TEST_main.cpp" line="30">
      <OverallResult success="true" durationInSeconds="0.005464"/>
    </TestCase>
    <OverallResults successes="4" failures="0" expectedFailures="0"/>
  </Group>
  <OverallResults successes="4" failures="0" expectedFailures="0"/>
</Catch>

[2019-11-12 5:27:28.158 PM Informational] Finished test: CmdLine.V8. Simple Display help
[2019-11-12 5:27:28.161 PM Informational] Finished Catch2Adapter test execution.
[2019-11-12 5:27:28.539 PM Informational] ========== Run finished: 1 tests run (0:00:00.7661416) ==========
`

**Comment by kervinpierre** [comment]
Just noticed that Github strips XML.

Here's a snapshot instead.

![image](https://user-images.githubusercontent.com/350213/68718348-f2711b00-0576-11ea-8b13-c6a8669b13e5.png)

**Comment by JohnnyHendriks** [comment]
The relevant information for this problem is during test discovery. Sadly, VS2019 had to be smart and decided it could skip this step as it is under the impression that it already has the information cached. The way around this is to switch configuration (_e.g._, from Release to Debug) and build the test again. This should trigger the discovery step. No need to actually run the test, the information for the discovery should appear in the test output after the build is finished.

**Comment by FAndersson** [comment]
Hi! I've also experienced the same problem, using Visual Studio version 16.3.9 and version 1.5.1 of the test adapter. For debugging I created a simple project with a single test file, and placed it directly in C:\, but navigating to the source code still doesn't work. So that seems to rule out your suggestion that the problem could be caused by long path names. I could upload the simple test project and/or my debug output from Visual Studio if that would help.

**Comment by kervinpierre** [comment]
Thanks for the explanation.  I didn't see any new discovery after switching targets. Just a message saying discovery isn't needed...
 
![image](https://user-images.githubusercontent.com/350213/69016824-35186600-0970-11ea-9184-3be92f283906.png)
![image](https://user-images.githubusercontent.com/350213/69016830-50837100-0970-11ea-976f-a6849605911a.png)

**Comment by kervinpierre** [comment]
...also get the discovery to run by clicking the "run all" above.  Output attached
[test-discovery.txt](https://github.com/JohnnyHendriks/TestAdapter_Catch2/files/3856356/test-discovery.txt)

**Comment by JohnnyHendriks** [comment]
Just released a new version of the test-adapter (1.6.0). It contains a number of improvements. Cannot reproduce this problem there (other than the unfortunate behavior of the Visual Studio Test Explorer). Not sure there is anything I can do about this, so I'm closing the issue.
